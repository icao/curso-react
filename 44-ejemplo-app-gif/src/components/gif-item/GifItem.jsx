import React from 'react'
import './GifItem.scss'

const GifItem = ({ title, url }) => {
  return (
    <div className='card'>
      <img src={url} alt={title} />
      <div className='card__title'>
        <h4>{title}</h4>
      </div>
    </div>
  )
}

export default GifItem
