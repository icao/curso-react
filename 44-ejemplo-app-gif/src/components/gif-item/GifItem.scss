@import '../../scss/Palette';

.card {
  // min-width: 150px;
  width: 300px;
  height: auto;
  border: 1px solid $color-grey-thin;
  border-radius: 5px;
  margin: 0.5rem;
  overflow: hidden;

  img {
    width: 100%;
    height: auto;
  }

  &__title {
    padding: 1rem;
    text-align: center;
    h4 {
      margin: 0;
    }
  }
}
