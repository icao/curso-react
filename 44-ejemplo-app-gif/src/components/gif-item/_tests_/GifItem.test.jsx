import { shallow } from 'enzyme'
import GifItem from '../GifItem'

describe('Probando componente <GifItem />', () => {
  const title = 'Soy el nombre del gif'
  const url = 'http://alguna_direccion_imagen'
  const wrapper = shallow(<GifItem title={title} url={url} />)

  test('<GifItem /> debe mostrarse correctamente y aceptar las props: title, url', () => {
    expect(wrapper).toMatchSnapshot()
  })

  test('El texto(titulo) renderizado debe ser el mismo a la prop titulo', () => {
    const titulo = wrapper.find('h4').text().trim()
    expect(titulo).toBe(title)
  })

  test("El atributo 'src' debe ser igual a la prop url", () => {
    const urlImg = wrapper.find('img').props().src
    expect(urlImg).toBe(url)
  })

  test("El atributo 'alt' igual a la prop titulo", () => {
    const altImg = wrapper.find('img').prop('alt')
    expect(altImg).toBe(title)
  })

  test('El componente debe de tener la clase animate__fadeInDown', () => {
    const classImg = wrapper.find('div').at(0).prop('className')
    console.log(classImg)
    // Se espera que exista la clase animate__fadeInDown en el div contenedor
    expect(classImg.includes('animate__fadeDown')).toBe(true)
  })
})
