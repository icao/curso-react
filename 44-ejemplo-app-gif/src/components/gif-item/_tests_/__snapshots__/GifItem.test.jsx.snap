// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Probando componente <GifItem /> <GifItem /> debe mostrarse correctamente y aceptar las props: title, url 1`] = `
<div
  className="card animate__animated animate__fadeInDown"
>
  <img
    alt="Soy el nombre del gif"
    src="http://alguna_direccion_imagen"
  />
  <div
    className="card__title"
  >
    <h4>
      Soy el nombre del gif
    </h4>
  </div>
</div>
`;
